mono/mini/mini-posix.c

```diff
@@ -1165,46 +1245,45 @@ mono_gdb_render_native_backtraces (pid_t crashed_pid)
 {
 #ifdef HAVE_EXECV
 	const char *argv [10];
-	FILE *commands;
-	char commands_filename [] = "/tmp/mono-gdb-commands.XXXXXX";
+	memset (argv, 0, sizeof (char*) * 10);
 
-	if (mkstemp (commands_filename) == -1)
-		return;
+	char commands_filename [100]; 
+	commands_filename [0] = '\0';
+	g_snprintf (commands_filename, sizeof (commands_filename), "/tmp/mono-gdb-commands.%d", crashed_pid);
```

Comment:
Why the need to change the format of the filename?

Reply:
The previous filename was using mkstemp, which allocates. Since I needed to pick a new way to generate a process-unique file (because we can just overwrite the file if it's guaranteed to be unique to the process), I chose the PID. 

I can go the other route of initializing a random seed somehow without allocating memory, and reading some PRNG bytes and write to the various file names in a loop until it works, but using the PID seemed more straightforward.  

It guaranteed we'll never have two monos trying to access the same on at once, which seems to be the only constraint on it.